### PR TITLE
Download both current and historic ISO currencies

### PR DESCRIPTION
**Pull request by anton-siardziuk**
see #8 : Add list of currencies that includes historical currencies

### CODE REVIEW

File: src/Fetcher.php [hunk 1]
```diff
@@ -10,27 +10,87 @@
     /**
      * @var
```
Comment (frederikbosch) [inline src/Fetcher.php]: Let's add `string` here.

File: src/Fetcher.php [hunk 2]
```diff
@@ -10,27 +10,87 @@
...
      */
-    private $location;
+    private $currentLocation;
+
+    /**
+     * @var
```
Comment (frederikbosch) [inline src/Fetcher.php]: Let's add `string` here.

File: src/Fetcher.php [hunk 3]
```diff
@@ -10,27 +10,87 @@
...
+        $this->fetch();
+
+        file_put_contents(
+            $fileName,
+            $serializer->serialize($this->historicCountries)
+        );
+    }
+
+    /**
+     * @param $fileName
```
Comment (frederikbosch) [inline src/Fetcher.php]: Let's add `string` here.

File: src/Fetcher.php [hunk 4]
```diff
@@ -10,27 +10,87 @@
...
+
+        file_put_contents(
+            $fileName,
+            $serializer->serialize($this->currentCountries)
+        );
+    }
 
     /**
-     * @param $location
+     * @param $fileName
```
Comment (frederikbosch) [inline src/Fetcher.php]: Let's add `string` here.

File: src/Fetcher.php [hunk 5]
```diff
@@ -10,27 +10,87 @@
...
-    private $countries;
+    private $currentCountries;
+
+    /**
+     * @var null|array|Country[]
+     */
+    private $historicCountries;
+
+    /**
+     * @param $currentLocation
```
Comment (frederikbosch) [inline src/Fetcher.php]: Let's add `string` here.

File: src/Fetcher.php [hunk 6]
```diff
@@ -10,27 +10,87 @@
...
+     * @param $historicLocation
```
Comment (frederikbosch) [inline src/Fetcher.php]: Let's add `string` here.

**Comment by frederikbosch** [comment]
Besides some docblock - which was not there before your PR but let's take the opportunity now to put it in - I have no comments. If you change the suggestions, I will merge it in and tag directly afterwards.

**Comment by anton-siardziuk** [comment]
Ok, np!

**Comment by anton-siardziuk** [comment]
@frederikbosch I have added type annotations to Fetcher, please check it out.

**Comment by frederikbosch** [comment]
Looks good!

**Comment by frederikbosch** [comment]
Merged and tagged. Thanks for your work! Now let's see what we can do on the `Money` side.

File: resources/historic.php [hunk 7]
```diff
@@ -0,0 +1,926 @@
...
+    'alphabeticCode' => 'ZAL',
+    'currency' => 'Financial Rand',
+    'minorUnit' => 0,
+    'numericCode' => 991,
+  ),
+  'LTL' => 
+  array (
+    'alphabeticCode' => 'LTL',
+    'currency' => 'Lithuanian Litas',
+    'minorUnit' => 0,
```
Comment (marcosdsanchez) [inline resources/historic.php:440]: Minor unit is not 0. It's 2. Seems that all historical currencies are set to 0.

**Review by marcosdsanchez — COMMENTED** [review]
All historical currencies have a `minorUnit` set to 0.

File: resources/historic.php (hunk 7 above)
Comment (frederikbosch) [reply resources/historic.php:440]: Can you do another PR?

Comment (marcosdsanchez) [reply resources/historic.php:440]: Unfortunately the information is not in the xmls that https://www.currency-iso.org provide. Is there any other trustful source of information that we can use?

Comment (anton-siardziuk) [reply resources/historic.php:440]: Wow, I haven't noticed it at all!

Comment (anton-siardziuk) [reply resources/historic.php:440]: What we can do here, is to fill this in by hand from Wiki page in some sort of XML and then parse it to all those formats. Those changes are quite rare, so might not be a bad idea I think to do this by hand.

Comment (frederikbosch) [reply resources/historic.php:440]: I think there also is a Symfony package that also busy with building a repository.

**Pull request merged**